src/vmm/src/cpu_config/x86_64/custom_cpu_template.rs

```diff
@@ -27,12 +29,18 @@ impl GetCpuTemplate for Option<CpuTemplateType> {
                 CpuTemplateType::Custom(template) => Ok(Cow::Borrowed(template)),
                 CpuTemplateType::Static(template) => {
                     let vendor_id = get_vendor_id_from_host().map_err(GetCpuVendor)?;
+                    let cpu_model = CpuModel::get_cpu_model();
                     match template {
                         StaticCpuTemplate::C3 => {
                             if &vendor_id != VENDOR_ID_INTEL {
                                 return Err(CpuVendorMismatched);
+                            } else if cpu_model != SKYLAKE_FMS
+                                && cpu_model != CASCADE_LAKE_FMS
+                                && cpu_model != ICE_LAKE_FMS
+                            {
+                                return Err(InvalidCpuModel);
```

Comment:
Mh, we're repeating these in quite a few places. Maybe we could have `StaticCpuTemplate::supported_cpu_models(&self) -> &'static [CpuModel]`, and then just a single check at the top of this huge match that goes `if !template.supported_cpu_models().contains(CpuModel::get_cpu_model()) { return Err(InvalidCpuModel) }`?

that could then also be used in the tests below, to avoid having to repeat the set of supported hardware all the time.